examples/headers/src/middleware.ts

```diff
@@ -17,32 +17,33 @@ const middlewares = {
 
       forward(response);
     },
-    async ({ request, forward }: MiddlewareFunctionProps) => {
-      const response = NextResponse.next();
+    async ({ request, response }: MiddlewareFunctionProps) => {
+      const _response = NextResponse.next();
 
       // Copy headers from the request to the response
       request.headers.forEach((value, key) => {
```

Comment:
Thanks for the quick response and update. I've tested v1.4.0, and the demo seems better. In testing, I found a minor issue. 

I believe this should be `request.headers` not `request.headers` as this doesn't apply the headers from the first chain function. 

v1.4.0
<img width="812" alt="Screenshot 2024-12-20 at 8 55 35 AM" src="https://github.com/user-attachments/assets/866433ad-dcdc-4655-833b-388c9427c610" />

What it should be:
<img width="812" alt="Screenshot 2024-12-20 at 8 53 36 AM" src="https://github.com/user-attachments/assets/98562688-aa79-491f-a6dc-3b289b2aa2ab" />

Reply:
This just shouldn’t work like you said - it will generate many issues with react server actions as it was before. It can also be an security issue as you can leak for example auth tokens as I mentioned before is issue that you’ve created.